etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/validation/ValidationRuleEngine.java

```diff
@@ -36,18 +35,17 @@ public void unloadRules() {
 
     @Override
     public void ensureRulesLoaded() throws RuleLoaderException {
+        // Double-checked locking - needed to protect from excessive sync locks
         if (rules.isEmpty()) {
             synchronized (this) {
                 if (rules.isEmpty()) {
-                    Path path =
-                            Paths.get(
-                                    getClass()
-                                            .getClassLoader()
-                                            .getResource(ruleDefinitionsFileName)
-                                            .getPath());
-
+                    InputStream resourceStream =
+                            getClass()
+                                    .getClassLoader()
+                                    .getResourceAsStream(ruleDefinitionsFileName);
+                    assert resourceStream != null;
```

Comment:
I think assertions run the risk of being disabled through JVM flags. Thoughts on using an if statement to check if an object is null?